github/resource_github_repository_deploy_key_test.go

```diff
@@ -11,21 +13,43 @@ import (
 	"github.com/hashicorp/terraform/terraform"
 )
 
+func TestSuppressDeployKeyDiff(t *testing.T) {
```

Comment:
I'm not really strongly opinionated on this, but typically I do 1 validation per test and typically the function name is quite verbose, almost describing the test circumstance. What is more important though is our unit test style is consistent, so if this is consistent with everything it's all good with me

Reply:
That's fair point, I'll break it apart.

Reply:
Actually we do have many unit tests (set aside acceptance tests which can have multiple steps) which have multiple test cases in them and I think it's fine? e.g. https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/validators_test.go#L12-L54

The thing which I agree needs changing though is the error message, so it's more obvious where each failure is coming from. 👍 I will fix that.